### PR TITLE
Connect App with API

### PR DESCRIPTION
**Pull request by DarkxDev**
- Make use of the Bookstore API as your backend.
- Used Axios with createAsyncThunk to make API calls.
- Fetch books from Bookstore API when the component has rendered with useEffect.

### CODE REVIEW

File: src/pages/Books.js [hunk 1]
```diff
@@ -10,28 +11,60 @@ const Books = () => {
...
+      dispatch(addBook(newBook));
+
+      // Clear the input fields
+      titleInput.value = '';
+      authorInput.value = '';
+    } catch (error) {
+      const errorMessageDiv = document.createElement('div');
+      errorMessageDiv.textContent = 'Error';
+      document.body.appendChild(errorMessageDiv);
+    }
```
Comment (anagudelogu) [inline src/pages/Books.js:50]: - [ ] You did a great job working on this milestone! 💯 However, according to the requirements, your `addBook` action should be refactored in order to handle API interaction.

image

If you look closely, this `trycatch` block is handling adding the book in the API, this is a perfect example to use an async thunk, you have a form that triggers a submit action that needs to update some data in your back and your client. Please consider moving this logic into a `createBook` thunk and handle the different states in your reducer! 🤓 🔧 

The end code could look something like this:

```js
const newBook = {
    item_id: itemId,
    title: titleInput.value,
    author: authorInput.value,
    category: 'Fiction',
};

// Note: The addBook now handles the interaction with the API
// and error handling
dispatch(addBook(newBook)      

// Clear the input fields
titleInput.value = '';
authorInput.value = '';   
```

File: src/pages/Books.js [hunk 2]
```diff
@@ -10,28 +11,60 @@ const Books = () => {
...
+        data: { item_id: itemId },
+      });
+
+      // Dispatch an action to remove the book with the given ID from the store
+      dispatch(removeBook(itemId));
+    } catch (error) {
+      const errorMessageDiv = document.createElement('div');
+      errorMessageDiv.textContent = 'Error';
+      document.body.appendChild(errorMessageDiv);
+    }
```
Comment (anagudelogu) [inline src/pages/Books.js:67]: - [ ] The same happens with the remove, as this requires asynchronous code, please consider using a thunk! 🤓 🔧 

```suggestion
  const handleRemoveBook = async (itemId) => {
    
      
      
        
      

      // Dispatch an action to remove the book with the given ID from the store
      dispatch(removeBook(itemId));
    
      
      
      
    
```

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @DarkxDev, 
I'm Andres.

While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @DarkxDev 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**